runtime/src/iree/hal/utils/libmpi_test.cc

```diff
@@ -13,40 +13,97 @@
 
 namespace {
 
-TEST(libmpi, DynamicLoadLibraryAndSymbols) {
-  iree_dynamic_library_t* library = NULL;
-  iree_hal_mpi_dynamic_symbols_t symbols = {0};
-  iree_status_t status =
-      iree_hal_mpi_library_load(iree_allocator_system(), &library, &symbols);
-  if (!iree_status_is_ok(status)) {
-    iree_status_fprint(stderr, status);
-    iree_status_ignore(status);
-    std::cerr << "Symbols cannot be loaded, skipping test.";
-    GTEST_SKIP();
+const int MPI_SUCCESS = 0;
+
+class LibmpiTest : public ::testing::Test {
+ protected:
+  static void SetUpTestSuite() {
+    iree_status_t status =
+        iree_hal_mpi_library_load(iree_allocator_system(), &library, &symbols);
+
+    if (!iree_status_is_ok(status)) {
+      iree_status_fprint(stderr, status);
+      iree_status_ignore(status);
+    } else {
+      EXPECT_EQ(symbols.MPI_Init(NULL, NULL), MPI_SUCCESS);
+    }
   }
 
-  IREE_EXPECT_OK(
-      MPI_RESULT_TO_STATUS(&symbols, MPI_Init(NULL, NULL), "MPI_Init"));
+  void SetUp() override {
+    if (!library) GTEST_SKIP() << "No MPI library available. Skipping suite.";
+    bool has_symbols = false;
+    unsigned char *syms_ptr = (unsigned char *)&symbols;
+    for (int i = 0; i < sizeof(symbols); i++) {
+      if (*(syms_ptr + i) != 0) {
+        has_symbols = true;
+        break;
+      }
+    }
```

Comment:
this isn't required as iree_dynamic_library_lookup_symbol fails if the symbol is not found and thus the entire library would fail to load - reinterpret casting the symbols here is dangerous as the struct may have additional non-pointer values.